### PR TITLE
[ARVIR] Adds support for moving the artwork using your finger, and adds a tap to place button

### PR DESCRIPTION
**Pull request by orta**
Video: https://ortastuff.s3.amazonaws.com/vids/eigen-ar.mp4

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">There were <a href="https://eigen-ci.s3.amazonaws.com/5016/index.html">snapshot errors</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_circle">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by orta** [comment]
OK, this is green now

File: Artsy/View_Controllers/ARVIR/ARAugmentedVIRInteractionController.m [hunk 1]
```diff
@@ -32,33 +33,33 @@ - (instancetype)initWithSession:(ARSession *)session config:(ARAugmentedRealityC
     _invisibleWalls = @[];
     _detectedPlanes = @[];
     _delegate = delegate;
+    CGRect bounds = [UIScreen mainScreen].bounds;
+    _pointOnScreenForArtworkProjection =CGPointMake(bounds.size.width/2, bounds.size.height/2);
```
Comment (ashfurrow) [inline Artsy/View_Controllers/ARVIR/ARAugmentedVIRInteractionController.m]: spacing here, not a huge deal. Prettier needs Objective-C support? 🤔

File: Artsy/View_Controllers/ARVIR/ARAugmentedVIRSetupViewController.m [hunk 2]
```diff
@@ -49,28 +48,80 @@ - (instancetype)initWithMovieURL:(NSURL *)movieURL config:(ARAugmentedRealityCon
...
+//  - Xth time without permission because they denied after the fact
+//
+
+- (NSString *)subtitleWithDefaults:(NSUserDefaults *)defaults hasPermission:(BOOL)hasPermission
+{
+    BOOL firstTime = ![defaults boolForKey:ARAugmentedRealityHasSeenSetup];
+    if (firstTime && hasPermission) {
+        return hasAccessSubtitle;
+
+    } else if(firstTime && !hasPermission) {
```
Comment (ashfurrow) [inline Artsy/View_Controllers/ARVIR/ARAugmentedVIRSetupViewController.m]: Spacing again.

File: Artsy/View_Controllers/ARVIR/ARAugmentedVIRSetupViewController.m [hunk 3]
```diff
@@ -49,28 +48,80 @@ - (instancetype)initWithMovieURL:(NSURL *)movieURL config:(ARAugmentedRealityCon
...
+    return hasDeniedAccessSubtitle;
+}
+
+- (NSString *)buttonTitleWithDefaults:(NSUserDefaults *)defaults hasPermission:(BOOL)hasPermission
+{
+    BOOL firstTime = ![defaults boolForKey:ARAugmentedRealityHasSeenSetup];
+    if (firstTime && hasPermission) {
+        return hasAccessButtonTitle;
+
+    } else if(firstTime && !hasPermission) {
```
Comment (ashfurrow) [inline Artsy/View_Controllers/ARVIR/ARAugmentedVIRSetupViewController.m]: And here.

File: Artsy/View_Controllers/ARVIR/ARAugmentedVIRSetupViewController.m [hunk 4]
```diff
@@ -49,28 +48,80 @@ - (instancetype)initWithMovieURL:(NSURL *)movieURL config:(ARAugmentedRealityCon
...
+
+- (SEL)buttonSelectorWithDefaults:(NSUserDefaults *)defaults hasPermission:(BOOL)hasPermission
+{
+    BOOL firstTime = ![defaults boolForKey:ARAugmentedRealityHasSeenSetup];
+    if (firstTime && hasPermission) {
+        return @selector(next);
+
+    } else if(firstTime && !hasPermission) {
+        return @selector(next);
+    }
```
Comment (ashfurrow) [inline Artsy/View_Controllers/ARVIR/ARAugmentedVIRSetupViewController.m]: These two can be simplified to just `if (firstTime)`, right?

**Pull request merged**